packages/app-desktop/tools/modifyReleaseAssets.ts

```diff
@@ -0,0 +1,96 @@
+
+import path = require('path');
+import { parseArgs } from 'util';
+import { Context, downloadFile, getTargetRelease, updateReleaseAsset, uploadReleaseAsset } from './githubReleasesUtils';
+import { GitHubRelease } from '../utils/checkForUpdatesUtils';
+import { GenerateInfo, generateLatestArm64Yml } from './generateLatestArm64Yml';
+
+const basePath = path.join(__dirname, '..');
+const downloadDir = path.join(basePath, 'downloads');
+
+// Renames release assets in Joplin Desktop releases
+const renameReleaseAssets = async (context: Context, release: GitHubRelease) => {
+	// Patterns used to rename releases
+	const renamePatterns: [RegExp, string][] = [
+		[/-arm64\.dmg$/, '-arm64.DMG'],
+	];
+
+	for (const asset of release.assets) {
+		for (const [pattern, replacement] of renamePatterns) {
+			if (asset.name.match(pattern)) {
+				const newName = asset.name.replace(pattern, replacement);
+				await updateReleaseAsset(context, asset.url, newName);
+
+				// Only rename a release once.
+				break;
+			}
+		}
+	}
+};
+
+// Creates release assets in Joplin Desktop releases
+const createReleaseAssets = async (context: Context, release: GitHubRelease) => {
+	// Create latest-mac-arm64.yml file and publish
+	let dmgPath;
+	let zipPath;
+	for (const asset of release.assets) {
+		if (asset.name.endsWith('arm64.zip')) {
+			zipPath = await downloadFile(asset, downloadDir);
+		} else if (asset.name.endsWith('arm64.DMG')) {
+			dmgPath = await downloadFile(asset, downloadDir);
+		}
+	}
+
+	const info: GenerateInfo = {
+		version: release.tag_name.slice(1),
+		dmgPath: dmgPath,
+		zipPath: zipPath,
+		releaseDate: new Date().toISOString(),
+	};
+
+	const latestArm64FilePath = generateLatestArm64Yml(info, downloadDir);
+	void uploadReleaseAsset(context, release, latestArm64FilePath);
+};
+
+
+const modifyReleaseAssets = async () => {
+	const args = parseArgs({
+		options: {
+			tag: { type: 'string' },
+			token: { type: 'string' },
+			repo: { type: 'string' },
+		},
+	});
+
+	console.log(args.values.tag);
+	console.log(args.values.token);
```

Comment:
Please remove at least this one since we wouldn't want to have our API token appear in the log.

Also consider prefixing these log statements so that we know what it is when looking at the log

Reply:
Can you please tell me what you mean by prefixing log statements? Do you mean instead of `console.log()` to use `console.info()/.error()` ?

Reply:
No I mean something like `console.log('GitHub tag:`, arg.values.tag)` etc. Otherwise it's difficult to know what it is in the log, or to search for it

Reply:
Ah ok. I deleted those lines since I was only using them for debugging purposes. The other console.log statements are more descriptive. 

Since I deleted the logic inside the afterAllArtifactBuild hook, the workflow shouldn't generate the file anymore